lib/http-assertions.js

```diff
@@ -0,0 +1,43 @@
+import { AssertionError } from 'zinnia:assert'
+
+/**
+ * @param {Response} res
+ * @param {string} [errorMsg]
+ */
+export async function assertOkResponse (res, errorMsg) {
+  if (res.ok) return
+
+  let body
+  try {
+    body = await res.text()
+  } catch {}
+  const err = new Error(`${errorMsg ?? 'Fetch failed'} (${res.status}): ${body?.trimEnd()}`)
+  err.statusCode = res.status
+  err.serverMessage = body
+  throw err
+}
+
+/**
+ * @param {Response} res
+ * @param {string} [errorMsg]
+ */
+export async function assertRedirectResponse (res, errorMsg) {
```

Comment:
Would be nice to add test coverage, but also not more than nice imo

Reply:
I'll think about this. Do you know if Fetch API `Response` object is easy to mock?

Reply:
It was easy :)